dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUIVideoCaptureSettings.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUIVideoCaptureSettings.g.h"
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    struct CameraCaptureUIVideoCaptureSettings : CameraCaptureUIVideoCaptureSettingsT<CameraCaptureUIVideoCaptureSettings>
+    {
+        CameraCaptureUIVideoCaptureSettings() = default;
+
+        void validate();
+
+        hstring GetFileExtension();
+        void Serialize(winrt::Windows::Foundation::Collections::ValueSet const& props);
+
+        wil::single_threaded_rw_property<bool> AllowTrimming{ true };
```

Comment:
Is it ok for these properties to be single-threaded (and therefore unsafe to use from multiple threads)?  cppwinrt classes are Agile by default.

Reply:
I don't think these single-threaded-property helpers can be used in this PR.  Your objects are not single-threaded so property reads/writes must be protected by a lock.  There is unfortunately not a mechanism to add a lock to the WIL property helpers so the property getter/setter methods need to be hand-authored with locking included.

Reply:
I've implemented property-level srwlock to ensure that reads and writes are properly protected. This means we can safely access these properties from multiple threads, now.